### PR TITLE
Implement meta-field key filtering

### PR DESCRIPTION
**Pull request by open-dynaMIX**
This commit implements a "meta_HasKey"-filter.

For this to work, the meta-field is now a "generic.GenericScalar()".

This implementation is up for discussion and only implemented for Form.meta atm. If we decide that this is the way to go, we can implement it for the other models as well.

### CODE REVIEW

File: caluma/form/filters.py [hunk 1]
```diff
@@ -11,6 +12,7 @@
 class FormFilterSet(FilterSet):
     search = SearchFilter(fields=("slug", "name", "description"))
     order_by = OrderingFilter(label="FormOrdering", fields=("name",))
+    meta__has_key = CharFilter(lookup_expr="has_key", field_name="meta")
```
Comment (winged) [inline caluma/form/filters.py]: I would get rid of the double underscore here, as it leaks out to the graphQL representation, and is derived from an implementation (django queryset filter syntax).

Comment (open-dynaMIX) [reply caluma/form/filters.py]: Done.

**Comment by open-dynaMIX** [comment]
Do we want this filter for all models with a meta-field?

File: caluma/form/filters.py (hunk 1 above)
Comment (czosel) [reply caluma/form/filters.py]: I agree, `meta_HasKey` looks quite funny on the API. This feature will always be restricted to `meta` fields, so some an alternative name might be `metaHasKey`.

I'm wondering though: Can we imagine a way to filter for things like "meta property X equals/is bigger than/is smaller than Y"? Potentially, for multiple properties at once? Maybe something like

`filterMeta: '{"X": { "operator": "equals", "value": "Y" }, "Z": {...}}'

**Comment by czosel** [comment]
I think this can be useful everywhere, yeah.

**Pull request merged**